ME2Randomizer/Classes/Randomizers/ME2/Misc/RTexts.cs

```diff
@@ -167,11 +172,324 @@ private static void UwuifyTalkFile(TalkFile tf, bool keepCasing, List<int> skipI
                 }
 
                 var str = sb.ToString();
-                str = str.Replace("fuck", keepCasing ? "UwU" : "uwu", StringComparison.InvariantCultureIgnoreCase);
+
+                if (addReactions)
+                {
+                    str = AddReactionToLine(strData, str, keepCasing);
+                }
+                else //reactions not enabled, use original uwuify UwU insertion
+                { 
+                    str = str.Replace("fuck", keepCasing ? "UwU" : "uwu", StringComparison.InvariantCultureIgnoreCase);
+                }
+
                 TLKHandler.ReplaceString(sref.StringID, str, langCode);
             }
         }
 
+        private static string AddReactionToLine(string vanillaLine, string modifiedLine, bool keepCasing)
+        {
+            string finalString = "";
+            bool dangerousLine = false;
+
+            if (modifiedLine.Length < 2 || Regex.Matches(modifiedLine, "[a-zA-Z]").Count == 0 || vanillaLine.Contains('{'))
```

Comment:
This will probably be faster if the regex is pre-compiled. Since this will run on a bazillion strings that will have a big impact on performance